cudax/test/launch/launch_smoke.cu

```diff
@@ -104,6 +104,44 @@ struct dynamic_smem_span
   }
 };
 
+struct launch_transform_to_int_convertible
+{
+  int value_;
+
+  struct int_convertible
+  {
+    cudaStream_t stream_;
+    int value_;
+
+    int_convertible(cudaStream_t stream, int value) noexcept
+        : stream_(stream)
+        , value_(value)
+    {
+      CHECK_FALSE(kernel_run_proof);
+    }
+
+    int_convertible(int_convertible&&) = delete;
+
+    ~int_convertible() noexcept
+    {
+      CUDART(cudaStreamSynchronize(stream_));
+      CHECK(kernel_run_proof);
+    }
+
+    using __launch_transform_result = int;
```

Comment:
If this type is a something that the user can use to indicate what type the kernel should accept, should this name not start with "__"? It gives an impression this is some internal thing that the user should not declare.

I guess the same comment for __cudax_launch_transform

Reply:
my thinking has been that `__launch_transform` would be an implementation detail of cudax. i am not anticipating a need for end users to customize it.